src/application/helpers.ts

```diff
@@ -426,6 +427,23 @@ const getPathParts = (path: string): Array<PathPart> => {
   return result;
 };
 
+export const isOptionalFieldsActive = (
+  sectionFields: ApplicationFormFields,
+): boolean | undefined => {
```

Comment:
The function name indicates a true of false response, undefined migth not be expected. So it would make sense that this doesn't return undefined

Reply:
Looking at the function overall, it seems a bit messy. For instance `optionalFieldsCheckbox` can be a field or undefined, yet it is cast later as a string. The purpose seems a bit lost to me.